### PR TITLE
Add AWS CN  S3 support

### PR DESCRIPTION
**Pull request by haitaopanhq**
Because aws-cn have to set env S3_URL, so I add some config list:  
1.   s3fs: add nonempt opts
2.  yaml/configmap: add S3_REGION S3_BUCKET S3_URL IAM_ROLE

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+all:
+	docker build -t artifact.onwalk.net/k8s/kube-s3:latest .
+	docker push artifact.onwalk.net/k8s/kube-s3:latest
```
Comment (freegroup) [inline Makefile]: don't use your own docker repo in a public pull request.

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -16,9 +16,9 @@ if [ "$IAM_ROLE" == "none" ]; then
   chmod 0400 /etc/passwd-s3fs
 
   echo 'IAM_ROLE is not set - mounting S3 with credentials from ENV'
-  /usr/bin/s3fs  ${S3_BUCKET} ${MNT_POINT} -d -d -f -o endpoint=${S3_REGION},allow_other,retries=5
+  /usr/bin/s3fs -o nonempty ${S3_BUCKET} ${MNT_POINT} -d -d -f -o endpoint=${S3_REGION},allow_other,retries=5 -o url=${S3_URL}
   echo 'started...'
 else
   echo 'IAM_ROLE is set - using it to mount S3'
-  /usr/bin/s3fs ${S3_BUCKET} ${MNT_POINT} -d -d -f -o endpoint=${S3_REGION},iam_role=${IAM_ROLE},allow_other,retries=5
+  /usr/bin/s3fs ${S3_BUCKET} -o nonempty ${MNT_POINT} -d -d -f -o endpoint=${S3_REGION},iam_role=${IAM_ROLE},allow_other,retries=5 -o url=${S3_URL}
```
Comment (gaul) [inline docker-entrypoint.sh:23]: Why add `-o nonempty`?  Repeated above.

File: docker-entrypoint.sh [hunk 3]
```diff
@@ -16,9 +16,9 @@ if [ "$IAM_ROLE" == "none" ]; then
   chmod 0400 /etc/passwd-s3fs
 
   echo 'IAM_ROLE is not set - mounting S3 with credentials from ENV'
-  /usr/bin/s3fs  ${S3_BUCKET} ${MNT_POINT} -d -d -f -o endpoint=${S3_REGION},allow_other,retries=5
+  /usr/bin/s3fs -o nonempty ${S3_BUCKET} ${MNT_POINT} -d -d -f -o endpoint=${S3_REGION},allow_other,retries=5 -o url=${S3_URL}
```
Comment (gaul) [inline docker-entrypoint.sh:19]: s3fs will exit if `-o url` is provided but the value is empty:

```
s3fs: option url has invalid format, missing http / https protocol
```

**Review by gaul — CHANGES_REQUESTED** [review]

**Comment by gaul** [comment]
Is this a Google project?  Is the CLA required?  :confused:

File: docker-entrypoint.sh (hunk 3 above)
Comment (haitaopanhq) [reply docker-entrypoint.sh:19]: S3_URL is env
for axample: http://s3.cn-north-1.amazonaws.com.cn

**Comment by haitaopanhq** [comment]
I will Resubmit with CLA email

File: docker-entrypoint.sh (hunk 3 above)
Comment (gaul) [reply docker-entrypoint.sh:19]: If the user does not provide a value, which most do not, and thus the value is empty then s3fs will fail with the error.  Did you test this?

**Pull request closed without merging**